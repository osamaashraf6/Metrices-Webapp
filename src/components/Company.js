import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import '../style/company.css';
import arrow from '../assets/arrow_circle_right.svg';
import bg from '../assets/bg.png';
import up from '../assets/arrow_up.svg';
import down from '../assets/arrow_down.svg';

const Company = ({
  symbol, name, change, price,
}) => (
  <div className="company-card">
    <div className="company-card-arrow-right">
      <NavLink to={`/${symbol}`}>
        <img src={arrow} alt="arrow-right" />
      </NavLink>
    </div>
    <div>
      <h2 className="company-card-title">
        {name}
        <small>
          (
          {symbol}
          )
        </small>
      </h2>
    </div>
    <div className="content">
      <h3>
        {price}
        {' '}
        $
      </h3>
      <h3>
        status:
        {' '}
        {change > 0 ? (
          <img src={up} alt="arrow-up" />
        ) : (
          <img src={down} alt="arrow-down" />
        )}
        {' '}
      </h3>
    </div>
    <div>
      <img className="company-img" src={bg} alt="bg" />
    </div>
  </div>
);

Company.propTypes = {
  symbol: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  change: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
};

export default Company;
