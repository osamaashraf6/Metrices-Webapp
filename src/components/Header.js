import React from 'react';
import { NavLink } from 'react-router-dom';
import mic from '../assets/mic-icon.svg';
import setting from '../assets/settings-icon.svg';
import '../style/header.css';
import leftArrow from '../assets/arrow_left.svg';

const Header = () => (
  <div className="header">
    <div className="header-logo">
      <NavLink to="/">
        <img src={leftArrow} alt="left arrow" />
      </NavLink>
      <h2>Stock Metrics</h2>
    </div>
    <div className="header-icons">
      <img src={mic} alt="mic-icon" />
      <img src={setting} alt="setting-icon" />
    </div>
  </div>
);

export default Header;
