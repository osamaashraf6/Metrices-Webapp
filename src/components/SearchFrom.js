import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { searchQuery } from '../redux/stock/stock';
import '../style/searchFrom.css';

const SearchFrom = () => {
  const dispatch = useDispatch();
  const { query } = useSelector((state) => state.stockReducer);
  return (
    <div className="form-search">
      <input
        value={query}
        className="search-input"
        placeholder="What you search for"
        onChange={(event) => dispatch(searchQuery(event.target.value.trim()))}
      />
    </div>
  );
};

export default SearchFrom;
