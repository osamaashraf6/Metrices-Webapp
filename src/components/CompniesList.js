import React from 'react';
import { useSelector } from 'react-redux';
import Company from './Company';
import Spinner from './Spinner';
import ErrorPage from '../pages/ErrorPage';
import '../style/companiesList.css';

const CompniesList = () => {
  const companies = useSelector((state) => state.stockReducer);
  const { loading, error } = companies;
  const searchQuery = companies.query;

  const filterDataSeacrh = companies.data.filter((comp) => {
    if (searchQuery === '') {
      return comp;
    }
    if (comp.name.toLowerCase().includes(searchQuery.toLowerCase())) {
      return comp;
    }
    return '';
  });

  if (error) {
    return (
      <>
        <ErrorPage />
      </>
    );
  }

  if (loading) {
    return (
      <>
        <Spinner />
      </>
    );
  }
  return (
    <>
      <div className="companies-list">
        {filterDataSeacrh.map(({
          symbol, name, change, price,
        }) => (
          <Company
            key={symbol}
            symbol={symbol}
            name={name}
            change={change}
            price={price}
          />
        ))}
      </div>
    </>
  );
};

export default CompniesList;
