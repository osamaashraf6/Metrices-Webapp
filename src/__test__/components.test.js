import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import CompniesList from '../components/CompniesList';
import Company from '../components/Company';
import SearchFrom from '../components/SearchFrom';
import Header from '../components/Header';

describe('Test react components', () => {
  it('header renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('Test CompniesList renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <CompniesList />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test Company renders correctly', () => {
  const item = {
    symbol: 'APPL',
    name: 'Apple',
    change: 1.5,
    price: 14,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Company
          symbol={item.symbol}
          name={item.name}
          change={item.change}
          price={item.price}
        />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test Search input renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <SearchFrom />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
