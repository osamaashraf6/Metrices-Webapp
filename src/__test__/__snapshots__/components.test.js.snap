// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Company renders correctly 1`] = `
<div
  className="company-card"
>
  <div
    className="company-card-arrow-right"
  >
    <a
      className=""
      href="/APPL"
      onClick={[Function]}
    >
      <img
        alt="arrow-right"
        src="arrow_circle_right.svg"
      />
    </a>
  </div>
  <div>
    <h2
      className="company-card-title"
    >
      Apple
      <small>
        (
        APPL
        )
      </small>
    </h2>
  </div>
  <div
    className="content"
  >
    <h3>
      14
       
      $
    </h3>
    <h3>
      status:
       
      <img
        alt="arrow-up"
        src="arrow_up.svg"
      />
       
    </h3>
  </div>
  <div>
    <img
      alt="bg"
      className="company-img"
      src="bg.png"
    />
  </div>
</div>
`;

exports[`Test CompniesList renders correctly 1`] = `
<div
  className="lds-hourglass spinner"
/>
`;

exports[`Test Search input renders correctly 1`] = `
<div
  className="form-search"
>
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="What you search for"
    value=""
  />
</div>
`;

exports[`Test react components header renders correctly 1`] = `
<div
  className="header"
>
  <div
    className="header-logo"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="left arrow"
        src="arrow_left.svg"
      />
    </a>
    <h2>
      Stock Metrics
    </h2>
  </div>
  <div
    className="header-icons"
  >
    <img
      alt="mic-icon"
      src="mic-icon.svg"
    />
    <img
      alt="setting-icon"
      src="settings-icon.svg"
    />
  </div>
</div>
`;
